blog/06-02-2023-catering-to-the-client-side.md

```diff
@@ -21,17 +21,24 @@ The key distinction between the client- and server-side feature flagging is the
 
-### Flag evaluation is slow
+### Network calls are slow
 
-With server-side flags, we can assume that evaluating a feature flag is a _relatively_ fast operation. In some scenarios a flagging decision can be made in-process (a very fast operation). If that's not possible then flag evaluation is typically a quick service call - akin to making a DB query or calling a remote cache.
+In the server, client processes might pull rule-sets for evaluating flags and use those as the basis for local flag evaluation.
+In other cases, flags might be evaluated out-of-process (by a remote service or some co-located process) and the results of the evaluation sent back to the client.
+Regardless, with  server-side flags we can assume that evaluating a feature flag is a _relatively_ fast operation, even in cases where evaluations take place out-of-process.
 
-This situation is quite different with client-side flags. It's typically not possible to evaluate a flagging decision locally. Local evaluation would require the full feature flagging ruleset to be exposed to the client, something many organizations are not comfortable with. Additionally, flag evaluation may require additional context that's simply not available on the client at the time of evaluation.
+This situation is quite different with client-side flags.
+In cases of local evaluation, a ruleset (or a collection of them) must be sent to the client, in bulk or piecemeal. In cases of remote evaluation, the client must contact a remote server with the relevant context and receive an evaluated result.
 
-This means that a client-side flag evaluation requires a backend call. Unfortunately we should also anticipate the latency for such a call to be slow, particularly if our users are behind a spotty internet connection. In fact, with a native mobile app we have to handle a fully disconnected client.
+This means that a client-side systems require network calls. Unfortunately we should also anticipate the latency for such a call to be slow, particularly if our users are behind a spotty internet connection. In fact, with a native mobile app we have to handle a fully disconnected client.
 
-### Pre-evaluation
+### Locally-evaluated systems and initialization
 
-So client-side flag evaluation is slow, but we've also seen that the inputs into that flag evaluation - the evaluation context - are fairly static for client-side apps, and that means the _results_ of flag evaluation are fairly static too.
+For systems wherein flags are evaluated locally, signalling readiness to the consuming application can be of particular importance. Until the ruleset governing flag evaluation is fetched, we can't guarantee accurate flag values for the given context. We need an API that supports events or other asynchronous mechanisms to communicate the state of the underlying system.
```

Comment:
I'd sort-of like to flesh this out a bit more, but I don't have time at the moment. We can either merge this as is and I will enhance it later, maybe with mermaid, or wait.